src/main/java/com/urbanairship/datacube/dbharnesses/HBaseDbHarness.java

```diff
@@ -477,6 +404,18 @@ private void flushBatch(Batch<T> batch) throws IOException, InterruptedException
             // of the pending writes, they should be removed from the batch so they are not retried later.
             // The operations that didn't get into the DB should be left in the batch to be retried later.
             log.warn("IOException when flushing batch to HBase", ioe);
+
+            if (ioe instanceof RetriesExhaustedWithDetailsException) {
+                RetriesExhaustedWithDetailsException details = (RetriesExhaustedWithDetailsException) ioe;
```

Comment:
Why do we need MDC over a more standard logger here?

Reply:
We don't. I felt that this was a lot of data to put into a string without additional structure (particularly the base64 encoded row key), especially since our current logging implementation surfaces the whole MDC.